tls/ttls.c

```diff
@@ -2246,6 +2246,8 @@ ttls_recv(void *tls_data, unsigned char *buf, size_t len, unsigned int *read)
 	case TTLS_MSG_CHANGE_CIPHER_SPEC:
 		/* Parsed as part of handshake FSM. */
 	case TTLS_MSG_HANDSHAKE:
+		if (len == 0)
+			return T_POSTPONE;
```

Comment:
Ok, we received only header of a handshake record, return here and having `TTLS_F_ST_HDRIV` in flags we won't read a record header again. Won't we be tricked if we receive a handshake record with zero length?

Reply:
> Won't we be tricked if we receive a handshake record with zero length?

As far as I understand, no.

`ttls_parse_record_hdr()` returns an error if `msglen` is less than `TTLS_HS_HDR_LEN` (which is 4). That's enough to read a handshake length. And then there is a check for `hslen` which returns an error if we have too short ClientHello message, that is less than 38 bytes. We also have a strict limit of at least 1 byte length in `ttls_hdr_check()`.